src/lib/blueprint.ts

```diff
@@ -50,17 +50,67 @@ export interface Endpoint {
   codeSamples: CodeSample[]
 }
 
-export interface Parameter {
+interface BaseParameter {
   name: string
   isRequired: boolean
   isUndocumented: boolean
   isDeprecated: boolean
   deprecationMessage: string
   description: string
-  jsonType: string
-  format: string
 }
 
+interface StringParameter extends BaseParameter {
+  format: 'string'
+  jsonType: 'string'
+}
+
+interface EnumParameter extends BaseParameter {
+  format: 'enum'
+  jsonType: 'string'
+  values: EnumValue[]
+}
+
+interface RecordParameter extends BaseParameter {
+  format: 'record'
+  jsonType: 'object'
+}
+
+interface ListParameter extends BaseParameter {
+  format: 'list'
+  jsonType: 'array'
```

Comment:
Copied this over from `ListProperty` but is `array` correct here? Because `typeof [] === 'object'`